### DISCUSSION TITLE
Raspberry Pi 4 running ubuntu 22.04 with Conbee II Stick

### DISCUSSION THREAD

**Discussion by hutchy71** [Koenkk/zigbee2mqtt#18402] [Devices]
With the above Raspberry I followed this Install Instruction:

[https://www.zigbee2mqtt.io/guide/installation/01_linux.html#optional-running-as-a-daemon-with-systemctl](url)

and get the following error-message

info  2023-07-22 23:26:02: Starting Zigbee2MQTT version 1.32.1 (commit #83205e4)
info  2023-07-22 23:26:02: Starting zigbee-herdsman (0.16.0)
error 2023-07-22 23:26:04: Error while starting zigbee-herdsman
error 2023-07-22 23:26:04: Failed to start zigbee
error 2023-07-22 23:26:04: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
error 2023-07-22 23:26:04: Exiting...
error 2023-07-22 23:26:04: Error: Error while opening serialport 'Error: Error Resource temporarily unavailable Cannot lock port'
    at SerialPort.<anonymous> (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/adapter/deconz/driver/driver.ts:200:28)
    at SerialPort._error (/opt/zigbee2mqtt/node_modules/@serialport/stream/dist/index.js:75:22)
    at /opt/zigbee2mqtt/node_modules/@serialport/stream/dist/index.js:111:18

while in  folder

/opt/zigbee2mqtt

running

`npm start`

I tried everything to fix the problem...couldn't solve it ...

**Comment by hutchy71** [comment]
My ztigbee2mqtt.services file looks like this

[Unit]
Description=zigbee2mqtt
After=network.target

[Service]
Environment=NODE_ENV=production
ExecStart=/usr/bin/npm start
WorkingDirectory=/opt/zigbee2mqtt
StandardOutput=null
StandardError=null
Restart=always
RestartSec=10s
User=my username

[Install]
WantedBy=multi-user.target



my configuratiuon.yaml:


homeassistant: false
permit_join: true
mqtt:
  base_topic: zigbee2mqtt
  server: mqtt://my_raspberrypi_ip:1881
serial:
  port: /dev/ttyACM0
advanced:
  network_key:
...keys---

frontend:
 
  port: 8181

  host: 0.0.0.0

`ls -l /dev/serial/by-id` look like this:
lrwxrwxrwx 1 root root 13 Jul 22 22:43 usb-dresden_elektronik_ingenieurtechnik_GmbH_ConBee_II_DE3019086-if00 -> ../../ttyACM0 

`ls -l /dev/ttyACM0` outcome:
crw-rw---- 1 root dialout 166, 0 Jul 22 23:45 /dev/ttyACM0

**Comment by lux73** [comment]
mak sure your ConBee Stick isn't actually used & blocked by other running Software like Phoscon or anything

make sure your User who wanted to run z2m (which should not be User 'root') is part of the Usergroup **dialout**

is this an newly installation? has it run anytime before?

mqtt Server up'n running? which?

btw before running & starting via daemon you should try to start manually via `npm start` inside your z2m Installation - after this it's time to start via daemon

**Reply by hutchy71** [reply]
I think Conbee II Stick is blocked, but I am unable to solve the problem.

My User is part of the group **dialout**.

Its a new installation. **It was running once, useing my Imac in the same network in two opened windows for _homebridge_ an _z2m_ http://localhost:port. After reboot the above mentioned error occured.**

Mosquitto mqtt Server ist installed and running on Raspberry Pi:

`systemctl status mosquitto`
● mosquitto.service - Mosquitto MQTT Broker
     Loaded: loaded (/lib/systemd/system/mosquitto.service; enabled; vendor preset: enab>
     Active: active (running) since Sat 2023-07-22 23:48:27 CEST; 11h ago
       Docs: man:mosquitto.conf(5)
             man:mosquitto(8)
    Process: 1828 ExecStartPre=/bin/mkdir -m 740 -p /var/log/mosquitto (code=exited, sta>
    Process: 1844 ExecStartPre=/bin/chown mosquitto /var/log/mosquitto (code=exited, sta>
    Process: 1849 ExecStartPre=/bin/mkdir -m 740 -p /run/mosquitto (code=exited, status=>
    Process: 1853 ExecStartPre=/bin/chown mosquitto /run/mosquitto (code=exited, status=>
    Process: 5313 ExecReload=/bin/kill -HUP $MAINPID (code=exited, status=0/SUCCESS)
   Main PID: 1856 (mosquitto)
      Tasks: 1 (limit: 8950)
     Memory: 2.4M
        CPU: 732ms
     CGroup: /system.slice/mosquitto.service
             └─1856 /usr/sbin/mosquitto -c /etc/mosquitto/mosquitto.conf

Jul 22 23:48:26 user-raspberry4 systemd[1]: Starting Mosquitto MQTT Broker...
Jul 22 23:48:27 user-raspberry4 systemd[1]: Started Mosquitto MQTT Broker.
Jul 23 11:32:23 user-raspberry4 systemd[1]: Reloading Mosquitto MQTT Broker...
Jul 23 11:32:23 user-raspberry4 systemd[1]: Reloaded Mosquitto MQTT Broker


`systemctl status homebridge`
● homebridge.service - Homebridge
     Loaded: loaded (/lib/systemd/system/homebridge.service; enabled; vendor preset: ena>
     Active: active (running) since Sat 2023-07-22 23:48:51 CEST; 11h ago
    Process: 2426 ExecStartPre=/bin/run-parts /etc/hb-service/homebridge/prestart.d (cod>
   Main PID: 4849 (hb-service)
      Tasks: 22 (limit: 8950)                                                            
     Memory: 248.4M                                                                      
        CPU: 21.102s                                                                     
     CGroup: /system.slice/homebridge.service                                            
             ├─4849 hb-service "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" >
             └─7041 homebridge "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" >
                                                                                         
Jul 22 23:48:32 user-raspberry4 systemd[1]: Starting Homebridge...
Jul 22 23:48:35 user-raspberry4 run-parts[3321]: Ensuring /var/lib/homebridge and /op>
Jul 22 23:48:51 user-raspberry4 systemd[1]: Started Homebridge.
Jul 23 11:32:28 user-raspberry4 start.sh[4849]: [23.7.2023, 11:32:28] [HB Supervisor]>
Jul 23 11:32:28 user-raspberry4 start.sh[4849]: [23.7.2023, 11:32:28] [HB Supervisor]>
Jul 23 11:32:28 user-raspberry4 start.sh[4849]: [23.7.2023, 11:32:28] [HB Supervisor]>

I'll give your hint, deleting the Daemon and starteing it manually a try, Is this correct?:

`sudo systemctl disable zigbee2mqtt.service && sudo rm /etc/systemd/system/zigbee2mqtt.service `

**Comment by hutchy71** [comment]
After removing the Daemon an starting manually this errors occurs:

/opt/zigbee2mqtt$ npm start

> zigbee2mqtt@1.32.1 start
> node index.js

Zigbee2MQTT:info  2023-07-23 12:13:37: Logging to console and directory: '/opt/zigbee2mqtt/data/log/2023-07-23.12-13-37' filename: log.txt
Zigbee2MQTT:info  2023-07-23 12:13:37: Starting Zigbee2MQTT version 1.32.1 (commit #83205e4)
Zigbee2MQTT:info  2023-07-23 12:13:37: Starting zigbee-herdsman (0.16.0)
Zigbee2MQTT:info  2023-07-23 12:13:39: zigbee-herdsman started (resumed)
Zigbee2MQTT:info  2023-07-23 12:13:39: Coordinator firmware version: '{"meta":{"maintrel":0,"majorrel":38,"minorrel":114,"product":0,"revision":"0x26720700","transportrev":0},"type":"ConBee2/RaspBee2"}'
Zigbee2MQTT:info  2023-07-23 12:13:39: Currently 0 devices are joined:
Zigbee2MQTT:warn  2023-07-23 12:13:39: `permit_join` set to  `true` in configuration.yaml.
Zigbee2MQTT:warn  2023-07-23 12:13:39: Allowing new devices to join.
Zigbee2MQTT:warn  2023-07-23 12:13:39: Set `permit_join` to `false` once you joined all devices.
Zigbee2MQTT:info  2023-07-23 12:13:39: Zigbee: allowing new devices to join.
Zigbee2MQTT:info  2023-07-23 12:13:40: Connecting to MQTT server at mqtt://192.168.178.33:1881
Zigbee2MQTT:error 2023-07-23 12:13:40: MQTT error: connect ECONNREFUSED 192.168.178.33:1881
Zigbee2MQTT:error 2023-07-23 12:13:40: MQTT failed to connect, exiting...
Zigbee2MQTT:info  2023-07-23 12:13:40: Stopping zigbee-herdsman...
Zigbee2MQTT:error 2023-07-23 12:13:41: MQTT error: connect ECONNREFUSED 192.168.178.33:1881
Zigbee2MQTT:info  2023-07-23 12:13:42: Stopped zigbee-herdsman




I changed the port in the configuratio.yaml from 1881 to 1883 and now it seems to work....

outcome:

Zigbee2MQTT:info  2023-07-23 12:21:47: Logging to console and directory: '/opt/zigbee2mqtt/data/log/2023-07-23.12-21-47' filename: log.txt
Zigbee2MQTT:info  2023-07-23 12:21:47: Starting Zigbee2MQTT version 1.32.1 (commit #83205e4)
Zigbee2MQTT:info  2023-07-23 12:21:47: Starting zigbee-herdsman (0.16.0)
Zigbee2MQTT:info  2023-07-23 12:21:49: zigbee-herdsman started (resumed)
Zigbee2MQTT:info  2023-07-23 12:21:49: Coordinator firmware version: '{"meta":{"maintrel":0,"majorrel":38,"minorrel":114,"product":0,"revision":"0x26720700","transportrev":0},"type":"ConBee2/RaspBee2"}'
Zigbee2MQTT:info  2023-07-23 12:21:49: Currently 0 devices are joined:
Zigbee2MQTT:warn  2023-07-23 12:21:49: `permit_join` set to  `true` in configuration.yaml.
Zigbee2MQTT:warn  2023-07-23 12:21:49: Allowing new devices to join.
Zigbee2MQTT:warn  2023-07-23 12:21:49: Set `permit_join` to `false` once you joined all devices.
Zigbee2MQTT:info  2023-07-23 12:21:49: Zigbee: allowing new devices to join.
Zigbee2MQTT:info  2023-07-23 12:21:51: Connecting to MQTT server at mqtt://192.168.178.33:1883
Zigbee2MQTT:info  2023-07-23 12:21:51: Connected to MQTT server
Zigbee2MQTT:info  2023-07-23 12:21:51: MQTT publish: topic 'zigbee2mqtt/bridge/state', payload 'online'
Zigbee2MQTT:info  2023-07-23 12:21:51: Started frontend on port 0.0.0.0:8181
Zigbee2MQTT:info  2023-07-23 12:21:51: MQTT publish: topic 'zigbee2mqtt/bridge/config', payload '{"commit":"83205e4","coordinator":{"meta":{"maintrel":0,"majorrel":38,"minorrel":114,"product":0,"revision":"0x26720700","transportrev":0},"type":"ConBee2/RaspBee2"},"log_level":"info","network":{"channel":11,"extendedPanID":"0xdddddddddddddddd","panID":6754},"permit_join":true,"version":"1.32.1"}'
Zigbee2MQTT:info  2023-07-23 12:21:51: Zigbee2MQTT started!

**Reply by lux73** [reply]
nice 👍🏻
